lxd/instance/drivers/driver_qemu.go

```diff
@@ -5761,6 +5748,16 @@ func (d *qemu) Info() instance.Info {
 }
 
 func (d *qemu) Metrics() (*metrics.MetricSet, error) {
+	val := d.expandedConfig["security.agent.metrics"]
+
+	if val == "" || shared.IsTrue(val) {
```

Comment:
Feels like we could benefit from a function called something like `d.agentMetricsEnable() bool` that contains this logic so we can use it here and in `renderState`.